src/librustdoc/html/static/css/rustdoc.css

```diff
@@ -2183,6 +2183,29 @@ details.toggle[open] > summary::after {
 	content: "Collapse";
 }
 
+details.toggle:not([open]) > summary .docblock {
+	max-height: calc(1.5em + 0.75em);
+	overflow-y: hidden;
+}
+details.toggle:not([open]) > summary .docblock::after {
+	content: '';
+	position: absolute;
+	bottom: 0;
+	left: 0;
+	right: 0;
+	pointer-events: none;
+	background: linear-gradient(
```

Comment:
I don't love the look of this. It looks a bit out of place compared to the generally clean style of rustdoc. In the case of tables and other things, it seems reasonable to just not show them rather than do a paywall-style blur of everything below the first row.

Reply:
@rfcbot concern gradient-looks-weird

I see what you mean there. To avoid this, I think you'd want to only have a summary if the first element is a paragraph or header, and then use `white-space: nowrap; overflow: hidden; text-overflow: ellipsis;`?

Reply:
I liked the blur effect but text ellipsis is fine too. :laughing:

Reply:
Problem with this approach is that the ellipsis only appears if the current sentence is cut:

![image](https://github.com/user-attachments/assets/b4afc247-f208-45e5-902e-bc506b57eef6)

expanded:

![image](https://github.com/user-attachments/assets/b9159760-23b8-4e51-973a-3409e7ada23a)

I'll try to check if we can still get the "..." in any case...